### ISSUE TITLE
Type checking error in npm module aws-cdk-lib

### ISSUE THREAD

**Issue by rinfield** [denoland/deno#16695]
https://www.npmjs.com/package/aws-cdk-lib

If no type check is performed, it can be executed successfully.
In Node & TypeScript environments, type checking does not cause errors.

```ts
// main.ts
import { App } from "npm:aws-cdk-lib";
new App();
```

```console
$ deno --version
deno 1.28.1 (release, x86_64-unknown-linux-gnu)
v8 10.9.194.1
typescript 4.8.3

$ deno run --check main.ts
Check file:///workspaces/cdk-deno/main.ts
error: TS2305 [ERROR]: Module '"file:///root/.cache/deno/npm/registry.npmjs.org/aws-cdk-lib/2.50.0/index.d.ts"' has no exported member 'App'.
import { App } from "npm:aws-cdk-lib";
         ~~~
    at file:///workspaces/cdk-deno/main.ts:7:10
```

### d.ts

```ts
// /root/.cache/deno/npm/registry.npmjs.org/aws-cdk-lib/2.50.0/index.d.ts
<...snip...>
export * from './core';
<...snip...>
```

```ts
// /root/.cache/deno/npm/registry.npmjs.org/aws-cdk-lib/2.50.0/core/index.d.ts
export * from './lib';
```

```ts
// /root/.cache/deno/npm/registry.npmjs.org/aws-cdk-lib/2.50.0/core/lib/index.d.ts"
<...snip...>
export * from './app';
<...snip...>
```

```ts
// /root/.cache/deno/npm/registry.npmjs.org/aws-cdk-lib/2.50.0/core/lib/app.d.ts"
<...snip...>
export declare class App extends Stage {
<...snip...>
```

**Comment by rinfield** [comment]
maybe related #16654 and resolved by #16656?:thinking:

**Comment by dsherret** [comment]
It's having trouble finding some dependencies.

```
deno check --remote main.ts
```

```ts
TS2498 [ERROR]: Module '"deno:///missing_dependency.d.ts"' uses 'export =' and cannot be used with 'export *'.
export * as triggers from './triggers';
                          ~~~~~~~~~~~~
    at file:///C:/Users/david/AppData/Local/deno/npm/registry.npmjs.org/aws-cdk-lib/2.50.0/index.d.ts:235:27

TS2305 [ERROR]: Module '"file:///C:/Users/david/AppData/Local/deno/npm/registry.npmjs.org/aws-cdk-lib/2.50.0/index.d.ts"' has no exported member 'App'.
import { App } from "npm:aws-cdk-lib";
         ~~~
    at file:///V:/scratch/index.ts:1:10

Found 235 errors.
```
